### PR TITLE
Check that master key is returned when querying device for first time

### PR DESCRIPTION
**Pull request by erikjohnston**
See https://github.com/matrix-org/synapse/pull/11234

### CODE REVIEW

**Comment by anoadragon453** [comment]
There seem to be a mix of tabs vs. spaces going on here.

**Comment by erikjohnston** [comment]
> There seem to be a mix of tabs vs. spaces going on here.

WHY WOULD YOU DO THIS TO ME VIM WHY :sob:

File: lib/SyTest/Homeserver.pm [hunk 1]
```diff
@@ -241,7 +241,7 @@ sub configure_logger
 
       loggers => {
          synapse => {
-            level => "INFO"
```
Comment (DMRobertson) [inline lib/SyTest/Homeserver.pm:244]: Was this just for temporary debugging?

**Comment by DMRobertson** [comment]
https://github.com/matrix-org/sytest/runs/4091934296?check_suite_focus=true Looks like #1164.
https://github.com/matrix-org/sytest/runs/4091934326?check_suite_focus=true and https://github.com/matrix-org/sytest/runs/4091934384?check_suite_focus=true look like #1044. Does your branch contain the change from #1154 ?

No idea what happened in https://github.com/matrix-org/sytest/runs/4091934438?check_suite_focus=true

File: lib/SyTest/Homeserver.pm (hunk 1 above)
Comment (erikjohnston) [reply lib/SyTest/Homeserver.pm:244]: Oh blast, I carefully didn't add this, and then clearly did a `git commit -a`.

File: tests/41end-to-end-keys/08-cross-signing.pl [hunk 2]
```diff
@@ -652,6 +652,21 @@
...
+         my ( $content ) = @_;
+
+         log_if_fail "key query content1", $content;
+
+         # Check that we do in fact see the master key when querying the
+         # devices.
+         assert_json_keys( $content->{master_keys}, $user2_id );
+         assert_json_keys( $content->{master_keys}->{$user2_id}, "keys");
+         assert_json_keys( $content->{master_keys}->{$user2_id}{keys},
+            "ed25519:nqOvzeuGWT/sRx3h7+MHoInYj3Uk2LD/unI9kDYcHwk");
```
Comment (DMRobertson) [inline tests/41end-to-end-keys/08-cross-signing.pl:667]: Should we also expect to see this in the second call to matrix_get_e2e_keys on +685?

**Review by DMRobertson — COMMENTED** [review]
Seems reasonable; one question

File: tests/41end-to-end-keys/08-cross-signing.pl (hunk 2 above)
Comment (erikjohnston) [reply tests/41end-to-end-keys/08-cross-signing.pl:667]: Yes we should. I suppose its worth checking that its still there

**Review by DMRobertson — APPROVED** [review]
Thanks!

**Pull request merged**